python/extract_eeg_bids_archive.py

```diff
@@ -0,0 +1,217 @@
+    # Check if the upload already exist (re-upload)
+    # ---------------------------------------------------------------------------------------------
+   
+    for eeg_archive_file in eeg_archives_list:
+        eeg_archive_filename = eeg_archive_file['UploadLocation']
+        eeg_archive_path = os.path.join(eeg_incoming_dir, eeg_archive_filename)
+
+        if s3_obj and eeg_incoming_dir.startswith('s3://'):
+            eeg_archive_local_path = os.path.join(tmp_dir, eeg_archive_filename)
+            try:    
+                s3_obj.download_file(eeg_archive_path, eeg_archive_local_path)
+            except Exception as err:
+                imaging_io_obj.log_error_and_exit(
+                    f"{eeg_archive_path} could not be downloaded from S3 bucket. Error was\n{err}",
+                    MISSING_FILES
+                )
+            else:
+                eeg_archive_path = eeg_archive_local_path
+
+        elif eeg_incoming_dir.startswith('s3://'):
+            imaging_io_obj.log_error_and_exit(
+                f"{eeg_incoming_dir} is a S3 path but S3 server connection could not be established.",
+                BAD_CONFIG_SETTING
+            )
+
+        # Uncompress archive in tmp location
+        eeg_collection_path = imaging_io_obj.extract_archive(eeg_archive_path, 'EEG', tmp_dir)
+        tmp_eeg_session_path = None
+        eeg_session_rel_path = None
+        modalities = None
+        for (root, dirs, files) in os.walk(eeg_collection_path):
+            if os.path.basename(os.path.normpath(root)).startswith('ses-'):
+                tmp_eeg_session_path = root
+                modalities = dirs
+
+                eeg_session_rel_path_re = re.search(r'sub-.+$', root)
+                if eeg_session_rel_path_re:
+                    eeg_session_rel_path = eeg_session_rel_path_re.group()
+                else:
+                    imaging_io_obj.log_error_and_exit(
+                        "Could not find a subject folder in the bids structure for .",
+                        MISSING_FILES
+                    )
+
+        if not tmp_eeg_session_path:
+            imaging_io_obj.log_error_and_exit(
+                "Could not find a session folder in the bids structure for .",
+                MISSING_FILES
+            )
+
+        for modality in modalities:
+            tmp_eeg_modality_path = os.path.join(tmp_eeg_session_path, modality)
+            s3_data_dir = config_db_obj.get_config("EEGS3DataPath")
+
+            if s3_obj and s3_data_dir and s3_data_dir.startswith('s3://'):
+                s3_data_eeg_modality_path = os.path.join(s3_data_dir, eeg_session_rel_path, modality)
+
+                """
+                If the suject/session/modality bids data already exists
+                on the destination folder, delete if first before
+                copying the data
+                """ 
+                s3_obj.delete_file(s3_data_eeg_modality_path)
```

Comment:
I delete in case the new submission does not include all the previously uploaded files. Let's say, for example, that a task was uploaded by mistake and needs to be deleted. If I do not delete the previous submission, I'm scared that the new one will be merged with the previous and that files to be deleted will be kept.